### ISSUE TITLE
unexpected edits returned from `textDocument/onTypeFormatting` mid-comment

### ISSUE THREAD

**Issue by siegel** [clangd/clangd#1077]
As in #1074 and #1076, I'm working on `textDocument/onTypeFormatting` support in BBEdit.

Here is a case in which the returned edits don't make any sense to me at all.

Open source file:

```c
#include <stdio.h>
#include <stdlib.h>

int
main (int argc, char **argv)
{
	printf("hello world");
	
	if (1) { printf("abcdef\n"); exit (0); }
 	//	this is a test this is another test
}
```

Place insertion point between "test" and "this" in the middle of line 10.

Press `Return` to break the comment.

Expected: would receive edits consistent with breaking the line comment in two and forming two line comments.

Actual: received a significant number of edits that don't make any sense (and which appear to completely mangle the text, when applied).

As with #1076, it's not clear to me whether the position parameter should reflect the location at which the typed character is inserted, or whether it should even be inserted at all.

**Logs**

```
2022-03-23 12:44:37.365: stderr output from server: V[12:44:37.365] <<< {"jsonrpc":"2.0","method":"textDocument/didChange","params":{"contentChanges":[{"range":{"end":{"character":20,"line":9},"start":{"character":20,"line":9}},"text":"\n"}],"textDocument":{"uri":"file:///Users/siegel/Desktop/LSPTest/main.c","version":2}}}

I[12:44:37.365] <-- textDocument/didChange

2022-03-23 12:44:37.365: stderr output from server: V[12:44:37.365] <<< {"id":3,"jsonrpc":"2.0","method":"textDocument/onTypeFormatting","params":{"ch":"\n","options":{"insertFinalNewline":false,"insertSpaces":false,"tabSize":4,"trimTrailingWhitespace":false},"position":{"character":20,"line":9},"textDocument":{"uri":"file:///Users/siegel/Desktop/LSPTest/main.c"}}}

I[12:44:37.365] <-- textDocument/onTypeFormatting(3)

2022-03-23 12:44:37.366: stderr output from server: I[12:44:37.365] --> reply:textDocument/onTypeFormatting(3) 0 ms
V[12:44:37.365] >>> {"id":3,"jsonrpc":"2.0","result":[{"newText":"\n\n        ","range":{"end":{"character":1,"line":8},"start":{"character":23,"line":6}}},{"newText":"\n          ","range":{"end":{"character":10,"line":8},"start":{"character":9,"line":8}}},{"newText":"\n          ","range":{"end":{"character":30,"line":8},"start":{"character":29,"line":8}}},{"newText":"","range":{"end":{"character":35,"line":8},"start":{"character":34,"line":8}}},{"newText":"\n        ","range":{"end":{"character":40,"line":8},"start":{"character":39,"line":8}}},{"newText":"\n        ","range":{"end":{"character":20,"line":9},"start":{"character":41,"line":8}}}]}
```

**System information**

Output of `clangd --version`:

```
Homebrew clangd version 13.0.1
Features: mac+xpc
Platform: arm64-apple-darwin21.4.0
```

Editor/LSP plugin: BBEdit 14.x development version

Operating system: macOS 12.3 (21E230)

**Comment by HighCommander4** [comment]
This looks like the same testcase and steps as #1074. The fix I posted there should address it.

**Comment by HighCommander4** [comment]
Closing this, please let me know if the fix for #1074 did not address it.
